atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/transaction/impl/SerializableTransactionManager.java

```diff
@@ -83,8 +83,10 @@ public void registerClosingCallback(Runnable closingCallback) {
      * This constructor is necessary for the InitializeCheckingWrapper. We initialize a dummy transaction manager and
      * use the delegate instead.
      */
+    // TODO(ssouza): it's hard to change the interface of STM with this.
+    // We should extract interfaces and delete this hack.
```

Comment:
Or have a `TransactionManagerConfig` object to wrap up a bunch of these parameters

Reply:
This won't fix the need for this constructor, which is the real reason I don't like the whole inheritance tree of SerTM. I think we're better without them, and I'm going to submit a PR to fix that soonish.